src/doc.d

```diff
@@ -2151,8 +2152,35 @@ extern (C++) void highlightText(Scope* sc, Dsymbols* a, OutBuffer* buf, size_t o
                 static __gshared const(char)* blankline = "$(DDOC_BLANKLINE)\n";
                 i = buf.insert(i, blankline, strlen(blankline));
             }
-            leadingBlank = 1;
             iLineStart = i + 1;
+            if (!leadingBlank)
+            {
+                leadingBlank = true;
+            }
+            else if (!inCode)
+            {
+                // Group two or more newlines into a paragraph break
+                auto scout = iLineStart;
+                for (; scout < buf.offset; ++scout)
+                {
+                    c = buf.data[scout];
+                    if (c == ' ' || c == '\t')
+                    {
+                        continue;
+                    }
+                    if (c == '\n')
+                    {
+                        iLineStart = scout + 1;
+                        continue;
+                    }
+                    // We got to a non-whitespace, time to insert the paragraph
+                    // break.
+                    static __gshared immutable ps =
```

Comment:
Is there any need for all those qualifiers in new D code? You're just referring to a string literal, use ordinary `immutable ps = "it";`.